File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indices = new HashMap<>();

        for (int i = 0; i < nums.length; i++){
            indices.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++){
            int sum = target - nums[i];
            if(indices.containsKey(sum) && indices.get(sum) != i){
                return new int[]{i, indices.get(sum)};
            }
        }
        return new int[0];

    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums) - 1
        res = nums[0]

        while l <= r:
            if nums[l] < nums[r]:
                res = min(nums[l], nums[r])
                break
            m = (l + r) // 2

            res = min(res,nums[m])

            if nums[m] >= nums[l]:
                l += 1
            else:
                r -= 1
        return res


